pkg/playback/http_playback.go

```diff
@@ -0,0 +1,660 @@
+	if !rr.cassetteLoaded {
+		w.WriteHeader(400)
+		fmt.Fprint(w, "No cassette is loaded.\n")
+		return
+	}
+
+	switch rr.mode {
+	case Record:
+		rr.httpRecorder.webhookHandler(w, r)
+	case Replay:
+		fmt.Println("Error: webhook endpoint should never be called in replay mode")
+	case Auto:
+		if rr.isRecordingInAutoMode {
+			rr.httpRecorder.webhookHandler(w, r)
+		} else {
+			fmt.Println("Error: webhook endpoint should never be called in replay mode")
+		}
+	}
+}
+
+func (rr *RecordReplayServer) loadCassetteHandler(w http.ResponseWriter, r *http.Request) {
+	filepathVals, ok := r.URL.Query()["filepath"]
+
+	if !ok {
+		w.WriteHeader(400)
+		fmt.Fprint(w, "\"filepath\" query param must be present.\n")
+		return
+	}
+
+	if len(filepathVals) > 1 {
+		fmt.Printf("Multiple \"filepath\" param values given, ignoring all except first: %v\n", filepathVals[0])
+	}
+
+	relativeFilepath := filepathVals[0]
+
+	if !strings.HasSuffix(strings.ToLower(relativeFilepath), ".yaml") {
+		w.WriteHeader(400)
+		fmt.Fprint(w, "\"filepath\" must specify a .yaml file.\n")
+		return
+	}
+
+	if filepath.IsAbs(relativeFilepath) {
+		w.WriteHeader(400)
+		fmt.Fprint(w, "\"filepath\" must be a relative filepath, you provided a absolute filepath.\n")
+		return
+	}
+
+	absoluteFilepath := filepath.Join(rr.cassetteDirectory, relativeFilepath)
+
+	var shouldCreateNewFile bool
+
+	switch rr.mode {
+	case Record:
+		fmt.Println("/pb/cassette/load: Recording to: ", absoluteFilepath)
+		shouldCreateNewFile = true
+	case Replay:
+		fmt.Println("/pb/cassette/load: Replaying from: ", absoluteFilepath)
+		shouldCreateNewFile = false
+	case Auto:
+		_, err := os.Stat(absoluteFilepath)
+		if os.IsNotExist(err) {
+			fmt.Println("/pb/cassette/load: Recording to: ", absoluteFilepath)
+			shouldCreateNewFile = true
+			rr.isRecordingInAutoMode = true
+		} else {
+			fmt.Println("/pb/cassette/load: Replaying from: ", absoluteFilepath)
+			shouldCreateNewFile = false
+			rr.isRecordingInAutoMode = false
+		}
+	}
+
+	if shouldCreateNewFile {
+		fileHandle, err := os.Create(absoluteFilepath)
+		if err != nil {
+			handleErrorInHandler(w, err, 500)
+		}
+
+		err = rr.httpRecorder.insertCassette(fileHandle)
+		if err != nil {
+			handleErrorInHandler(w, err, 500)
+		}
+	} else {
+		fileHandle, err := os.Open(absoluteFilepath)
+		if err != nil {
+			handleErrorInHandler(w, err, 500)
+		}
+
+		err = rr.httpReplayer.readCassette(fileHandle)
+		if err != nil {
+			handleErrorInHandler(w, err, 500)
+		}
+	}
+
+	rr.cassetteLoaded = true
+}
+
+// InitializeServer sets up and returns a http.Server that acts as a playback proxy
+func (rr *RecordReplayServer) InitializeServer(address string) *http.Server {
+	customMux := http.NewServeMux()
+	server := &http.Server{Addr: address, Handler: customMux}
+
+	// --- Webhook endpoint
+	customMux.HandleFunc("/pb/webhooks", rr.webhookHandler)
+
+	// --- Server control handlers
+	customMux.HandleFunc("/pb/mode/", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
I would factor the "model" logic out from the "controller" logic in these handlers.

i.e. you'd have
```
func (rr *RecordReplayServer) SwitchMode (modeString string) (bool, string) {...}
func (rr *RecordReplayServer) SetRoot (absoluteCassetteDir string) (bool, string) {...}
func (rr *RecordReplayServer) LoadCassette (filepath string) (bool, string) {...}
```
(where bool is "success" and string is "message"), also for throroughness you should probably be passing the error message through in the response body as well as to stdout -- might help users get the debug output they need in their test output

Reply:
Made this refactor in the latest commit, though slightly different from what you outlined (instead of returning bool & string, I just return error). 

The consequence of this tweak is that the "model" can't specify a success message, so it falls to the controller to specify the success message when it handles the (err == nil) case. This feels like an okay division of responsibility to me.